### PR TITLE
docs:#1092 housekeeping for adjacent segment merging

### LINKED ISSUE #1092: Explain that we now do adjacent segment merging in Tiered Storage docs
### Describe the Issue

https://github.com/redpanda-data/core-internal/issues/130 changes a fairly important underlying behavior of how we manage data in customers' S3/ABS/GCS buckets. Users should be aware of this when they read up on Tiered Storage, just so they know its happening: https://docs.redpanda.com/docs/manage/tiered-storage/

A few sentences explaining it's happening and that it improves scalability and performance  are warranted.

@jcsp  is the SME here

### PR DESCRIPTION
**Pull request by micheleRP**
resolves #1092 

preview
- https://deploy-preview-1207--redpanda-documentation.netlify.app/docs/beta/manage/tiered-storage/#cloud-storage-housekeeping
- https://deploy-preview-1207--redpanda-documentation.netlify.app/docs/beta/reference/tunable-properties/#cloud-storage-properties

### CODE REVIEW

**Comment by micheleRP** [comment]
Closed #1200 and moved only changes to 23.1 into this PR.

**Comment by micheleRP** [comment]
@Lazin Do `cloud_storage_segment_size_target` and `cloud_storage_segment_size_min` require restart?

File: docs/manage/tiered-storage.mdx [hunk 1]
```diff
@@ -407,6 +435,11 @@ You may want to configure the following properties:
 | `cloud_storage_max_connection_idle_time_ms` | The maximum idle time for persistent HTTP connections. Differs depending on the cloud provider. Default is 5000 ms, which is sufficient for most providers. |
 | `cloud_storage_segment_max_upload_interval_sec` | Sets the number of seconds for idle timeout. If this property is empty, Redpanda uploads metadata to the cloud storage, but the segment is not uploaded until it reaches the `segment.bytes` size. By default, the property is empty. |
 | `cloud_storage_cache_check_interval` | The time, in milliseconds, between cache checks. The size of the cache can grow quickly, so it’s important to have a small interval between checks, but if the checks are too frequent, they consume a lot of resources. Default is 30000 ms. |
+| `cloud_storage_idle_timeout_ms` | The width of the sliding window for the moving average algorithm that calculates cloud storage utilization. Default is 10 seconds. |
```
Comment (Feediver1) [inline docs/manage/tiered-storage.mdx:438]: Are these properties in alphabetical order? If so this one coming after cloud_storage_upload... seems out of order.

File: docs/reference/tunable-properties.mdx [hunk 2]
```diff
@@ -330,6 +330,60 @@ Maximum backoff interval when there is nothing to upload for a partition.
 
 **Restart required**: yes
 
+---
+
+### cloud_storage_idle_timeout_ms
```
Comment (Feediver1) [inline docs/reference/tunable-properties.mdx:335]: this needs to go in alphabetical order in this file, as do the following ones.

File: docs/reference/tunable-properties.mdx [hunk 3]
```diff
@@ -330,6 +330,60 @@ Maximum backoff interval when there is nothing to upload for a partition.
...
+
+---
+
+### cloud_storage_segment_size_target
+
+The desired segment size in cloud storage. The default segment size is controlled by `log_segment_size` and the `segment.bytes` topic configuration property. This property can be set to a value larger than this default segment size, but because that triggers a lot of segment reuploads, it's not recommended.
+
+**Default**: the local segment size for the topic
+
+**Restart required**:
```
Comment (Feediver1) [inline docs/reference/tunable-properties.mdx:375]: yes or no?

File: docs/reference/tunable-properties.mdx [hunk 4]
```diff
@@ -330,6 +330,60 @@ Maximum backoff interval when there is nothing to upload for a partition.
...
+
+---
+
+### cloud_storage_segment_size_min
+
+The smallest segment size in the cloud that Redpanda keeps.
+
+**Default**: 50% of log segment size
+
+**Restart required**:
```
Comment (Feediver1) [inline docs/reference/tunable-properties.mdx:385]: yes or no?

**Review by Feediver1 — APPROVED** [review]
See the comments about the order of the properties, and what about yes or no in those last two props?

File: docs/reference/tunable-properties.mdx (hunk 2 above)
Comment (micheleRP) [reply docs/reference/tunable-properties.mdx]: I've alphabetized them now!

**Comment by micheleRP** [comment]
Hi @Lazin. Do cloud_storage_segment_size_target and cloud_storage_segment_size_min require restart? And after I include the answer to that, is this OK to merge in our Beta docs?

File: docs/reference/tunable-properties.mdx (hunk 3 above)
Comment (micheleRP) [reply docs/reference/tunable-properties.mdx]: properties are being added/verified in https://github.com/redpanda-data/documentation/pull/1203

File: docs/reference/tunable-properties.mdx (hunk 4 above)
Comment (micheleRP) [reply docs/reference/tunable-properties.mdx]: properties are being added/verified in https://github.com/redpanda-data/documentation/pull/1203

**Comment by micheleRP** [comment]
@Lazin, @jcsp: I removed the properties here because Edward's updating all properties, including these ones, with https://github.com/redpanda-data/documentation/pull/1203. Please let me know if this needs any changes. Hoping to merge this by Wednesday!

**Review by Lazin — APPROVED** [review]

File: docs/manage/tiered-storage.mdx [hunk 5]
```diff
@@ -372,6 +372,34 @@ Redpanda uses the following properties to configure transport. Edit them with th
 | `cloud_storage_segment_max_upload_interval_sec` | The number of seconds for idle timeout. If this property is empty, Redpanda uploads metadata to the cloud storage, but the segment is not uploaded until it reaches the `segment.bytes` size. By default, the property is empty. |
 | `cloud_storage_trust_file` | The public certificate used to validate the TLS connection to cloud storage. If this is empty, Redpanda uses your operating system's CA cert pool. |
 
+## Cloud storage housekeeping
```
Comment (mattschumpert) [inline docs/manage/tiered-storage.mdx:375]: This section pedantically goes through how we calculate when the housekeeping is triggered without first defining what housekeeping does.  Let's say what housekeeping IS first.

File: docs/manage/tiered-storage.mdx [hunk 6]
```diff
@@ -372,6 +372,34 @@ Redpanda uses the following properties to configure transport. Edit them with th
...
+
+To improve performance and scalability, Redpanda performs cloud storage housekeeping when the system is idle. Redpanda constantly calculates cloud storage utilization using the moving average with a sliding window algorithm. The width of the sliding window is defined by the `cloud_storage_idle_timeout_ms` property, which has a default of 10 seconds. 
+
+If the utilization (requests per second) drops below the threshold, then cloud storage is considered idle, and cloud storage housekeeping begins. The threshold is defined by the `cloud_storage_idle_threshold_rps` property, which has a default of one request per second. Cloud storage is considered idle if, during the last 10 seconds, there were 10 or less cloud storage API requests.
+
+If the cloud storage API becomes active after housekeeping begins, then housekeeping is paused until it becomes idle again. If cloud storage is not idle for `cloud_storage_housekeeping_interval_ms milliseconds`, then housekeeping is forced to run until it completes. This guarantees that all housekeeping jobs are run once for every `cloud_storage_housekeeping_interval_ms`. 
+
+### Adjacent segment merging
```
Comment (mattschumpert) [inline docs/manage/tiered-storage.mdx:383]: @jcsp  after reading the description, should we not  just call this feature 'cloud storage housekeeping', 'small segment merging' or 'small segment compaction'?  the adjacent part really isn't the point. To me this is housekeeping.  thoughts?

File: docs/manage/tiered-storage.mdx (hunk 5 above)
Comment (micheleRP) [reply docs/manage/tiered-storage.mdx:375]: > This section pedantically goes through how we calculate when the housekeeping is triggered without first defining what housekeeping does. Let's say what housekeeping IS first.

@mattschumpert Do you mean to jump right into segment merging? I thought adjacent segment merging is one part of cloud storage housekeeping, and that other aspects of housekeeping would be coming soon. Should we reposition this doc as just segment merging?

Comment (mattschumpert) [reply docs/manage/tiered-storage.mdx:375]: at least define it before this super lengthy description about the algorithm calculating when it runs.

I think this housekeeping refers mainly to adjacent segment merging but @jcsp  would have to confirm

**Review by mattschumpert — APPROVED** [review]

File: docs/manage/tiered-storage.mdx (hunk 5 above)
Comment (micheleRP) [reply docs/manage/tiered-storage.mdx:375]: Thanks @mattschumpert. I updated the intro: https://deploy-preview-1207--redpanda-documentation.netlify.app/docs/beta/manage/tiered-storage/#cloud-storage-housekeeping. I'll merge this, and if we have more edits, we'll open a new issue!

File: docs/manage/tiered-storage.mdx (hunk 6 above)
Comment (micheleRP) [reply docs/manage/tiered-storage.mdx:383]: @jcsp I need to merge this PR. Please let me know if you want changes to the feature name or description, and we'll open a new issue!

**Pull request merged**